test/PrintAsObjC/availability.swift

```diff
@@ -242,29 +354,61 @@
     @nonobjc func methodNotAvailableToObjC() {}
   
     @available(*, deprecated, renamed: "methodNotAvailableToObjC()")
-    @objc public func methodRenamedToMethodNotAvailableToObjC() {}
+    @objc public func deprecatedMethodRenamedToMethodNotAvailableToObjC() {}
+    @available(macOS, deprecated, renamed: "methodNotAvailableToObjC()")
+    @objc public func deprecatedOnMacOSMethodRenamedToMethodNotAvailableToObjC() {}
+
+    @available(*, unavailable, renamed: "methodNotAvailableToObjC()")
+    @objc public func unavailableMethodRenamedToMethodNotAvailableToObjC() {}
+    @available(macOS, unavailable, renamed: "methodNotAvailableToObjC()")
+    @objc public func unavailableOnMacOSMethodRenamedToMethodNotAvailableToObjC() {}
 
     @available(*, deprecated, renamed: "simpleProperty")
-    @objc public func methodRenamedToSimpleProperty() {}
+    @objc public func deprecatedMethodRenamedToSimpleProperty() {}
+    @available(macOS, deprecated, renamed: "simpleProperty")
+    @objc public func deprecatedOnMacOSMethodRenamedToSimpleProperty() {}
+
+    @available(*, unavailable, renamed: "simpleProperty")
+    @objc public func unavailableMethodRenamedToSimpleProperty() {}
+    @available(macOS, unavailable, renamed: "simpleProperty")
+    @objc public func unavailableOnMacOSMethodRenamedToSimpleProperty() {}
 
     @objc(methodReturningInt) public func simpleMethodReturningInt() -> Int { return -1 }
   
-    @available(*, deprecated, renamed: "methodWithoutCustomObjCName(value:)")
-    @objc public func methodRenamedToMethodWithouCustomObjCName(value: Int) -> Int { return -1 }
     @objc public func methodWithoutCustomObjCName(value: Int) -> Int { return -1 }
 
+    @available(*, deprecated, renamed: "methodWithoutCustomObjCName(value:)")
+    @objc public func deprecatedMethodRenamedToMethodWithouCustomObjCName(value: Int) -> Int { return -1 }
```

Comment:
Not that it's super important, but for this family you wrote "Withou" instead of "Without".

Reply:
@jrose-apple Should I fix it here or can leave it for the next PR?

Reply:
I think it's okay to leave it. Let's get this merged!